test/OpenTelemetry.Tests/Logs/LogRecordTest.cs

```diff
@@ -107,6 +107,74 @@ public void CheckStateForUnstructuredLogWithStringInterpolation()
             Assert.Equal(message.ToString(), state.ToString());
         }
 
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1602:Enumeration items should be documented", Justification = "enum for test")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.OrderingRules", "SA1201:Elements should appear in the correct order", Justification = "enum for test")]
+        public enum Field
+        {
+            FormattedMessage,
+            State,
+            StateValues,
+        }
+
+        [Fact]
+        public void StateCanBeSetByProcessor()
+        {
+            var exportedItems = new List<LogRecord>();
+            using var loggerFactory = LoggerFactory.Create(builder => builder
+                .AddOpenTelemetry(options =>
+                {
+                    options.AddProcessor(new RedactionProcessor(exportedItems, Field.State));
+                }));
+
+            var logger = loggerFactory.CreateLogger<LogRecordTest>();
+            {
+                logger.LogInformation($"This should be replaced {0.99}.");
+
+                var state = exportedItems[0].State as IReadOnlyList<KeyValuePair<string, object>>;
+                Assert.Equal("newStateKey", state[0].Key.ToString());
+                Assert.Equal("newStateValue", state[0].Value.ToString());
+            }
+        }
+
+        [Fact]
+        public void StateValuesCanBeSetByProcessor()
+        {
+            var exportedItems = new List<LogRecord>();
+            using var loggerFactory = LoggerFactory.Create(builder => builder
+                .AddOpenTelemetry(options =>
+                {
+                    options.AddProcessor(new RedactionProcessor(exportedItems, Field.StateValues));
+                }));
+
+            var logger = loggerFactory.CreateLogger<LogRecordTest>();
+            logger.Log(
+                LogLevel.Information,
+                0,
+                new List<KeyValuePair<string, object>> { new KeyValuePair<string, object>("OldKey", "OldValue") },
+                null,
+                (s, e) => "OpenTelemetry!");
+
+            var stateValue = exportedItems[0];
+            Assert.Equal(new KeyValuePair<string, object>("newStateValueKey", "newStateValueValue"), stateValue.StateValues[0]);
+        }
+
+        [Fact]
+        public void FormattedMessageCanBeSetByProcessor()
+        {
+            var exportedItems = new List<LogRecord>();
+            using var loggerFactory = LoggerFactory.Create(builder => builder
+                .AddOpenTelemetry(options =>
+                {
+                    options.AddProcessor(new RedactionProcessor(exportedItems, Field.FormattedMessage));
+                }));
+
+            var exporterLogger = loggerFactory.CreateLogger<LogRecordTest>();
```

Comment:
Maybe just call it "logger"?

```suggestion
            var logger = loggerFactory.CreateLogger<LogRecordTest>();
```